File: dart-pub/utm_parse/lib/src/utm_params.dart
/// UTM Parameters container
class UtmParams {
  final String? source;
  final String? medium;
  final String? campaign;
  final String? term;
  final String? content;
  final String? platform;

  const UtmParams({
    this.source,
    this.medium,
    this.campaign,
    this.term,
    this.content,
    this.platform,
  });

  bool get isEmpty =>
      source == null &&
      medium == null &&
      campaign == null &&
      term == null &&
      content == null;

  @override
  String toString() =>
      'UtmParams(source: $source, medium: $medium, campaign: $campaign)';

  Map<String, dynamic> toMap() {
    return {
      if (source != null) 'source': source,
      if (medium != null) 'medium': medium,
      if (campaign != null) 'campaign': campaign,
      if (term != null) 'term': term,
      if (content != null) 'content': content,
      if (platform != null) 'platform': platform,
    };
  }
}

File: .net/TanStackShip.UtmParse/UtmParams.cs
namespace TanStackShip.UtmParse;

/// <summary>
/// UTM Parameters container
/// </summary>
public class UtmParams
{
    /// <summary>utm_source - the referrer</summary>
    public string? Source { get; set; }

    /// <summary>utm_medium - marketing medium</summary>
    public string? Medium { get; set; }

    /// <summary>utm_campaign - campaign name</summary>
    public string? Campaign { get; set; }

    /// <summary>utm_term - paid search keyword</summary>
    public string? Term { get; set; }

    /// <summary>utm_content - ad content</summary>
    public string? Content { get; set; }

    /// <summary>Detected platform from click ID</summary>
    public string? Platform { get; set; }

    /// <summary>Check if all UTM params are empty</summary>
    public bool IsEmpty => 
        Source == null && 
        Medium == null && 
        Campaign == null && 
        Term == null && 
        Content == null;

    public override string ToString() =>
        $"UtmParams(Source={Source}, Medium={Medium}, Campaign={Campaign})";
}
